src/pages/api/articles/[id].tsx

```diff
@@ -1,11 +1,18 @@
 import { articles } from '@/data';
+import { NextApiResponse } from 'next';
+import { Article } from '@/src/types';
 
-export default ({ query: { id } }, res) => {
-  const articleFound = articles.find((article) => article.id === id); // Article | undefined
+export default (
+  { query: { id } }: { query: { id: string } },
+  res: NextApiResponse
+): void => {
+  const articleFound = articles.find(
+    (article: Article) => article.id === id
```

Comment:
ここはこう
```suggestion
    article => article.id === Number(id)
```
ここも型推論効くのと、`id` はパスパラメータだから string 型になっちゃってるから `number`にキャストしないといけない

Reply:
今stringになってない？

```typescript
export type Article = {
  id: string;
  image: string;
  title: string;
  content: string;
  createdAt: string;
  likes: number;
};
```

Reply:
とりあえず型推論されてる場所の変更はした！